### PR TITLE
MIGENG-684 added 5 new MTV fields , and tests

### PR DESCRIPTION
**Pull request by jonathanvila**
This PR covers : 
https://issues.redhat.com/browse/MIGENG-684 ( vm_drs_config )
https://issues.redhat.com/browse/MIGENG-685 ( vm_ha_config )
https://issues.redhat.com/browse/MIGENG-686 ( ballooned_memory )
https://issues.redhat.com/browse/MIGENG-688 ( encrypted_disk )
( opaque_network )

Also has a refactoring on the check on **Utils.verifyRulesFiredNames** , to not check the rules firing in a particular order , as we can not guarantee that as Drools has a random order (except if you use "salience" ) execution

NOTE : not mergeable until the values for the new flags have been inserted in the database
https://issues.redhat.com/browse/MIGENG-826

### CODE REVIEW

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java [hunk 1]
```diff
@@ -22,6 +22,12 @@
     public static final String SHARED_DISK_FLAG_NAME = "Shared Disk";
     public static final String CPU_MEMORY_HOTPLUG_FLAG_NAME = "CPU/Memory hotplug";
     public static final String CPU_AFFINITY_FLAG_NAME = "CPU Affinity";
+    public static final String VM_DRS_CONFIG_FLAG_NAME = "VMWare DRS";
+    public static final String VM_HA_CONFIG_FLAG_NAME = "VM HA";
+    public static final String BALLOONED_MEMORY_FLAG_NAME = "Balloneed memory";
```
Comment (m-brophy) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java]:
```suggestion
    public static final String BALLOONED_MEMORY_FLAG_NAME = "Ballooned memory";
```
Just a typo

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java [hunk 2]
```diff
@@ -374,6 +374,93 @@ public void test_CpuAffinityNotNull()
...
+        vmWorkloadInventoryModel.setHasEncryptedDisk(true);
+        vmWorkloadInventoryModel.setHasOpaqueNetwork(true);
+        facts.put("vmWorkloadInventoryModel", vmWorkloadInventoryModel);
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = new WorkloadInventoryReportModel();
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
+
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+        Assert.assertEquals(2, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener, "AgendaFocusForTest", "Flag_Opaque_Network", "Flag_Encrypted_Disk", "Flag_Ballooned_Memory", "Flag_VM_HA_Config", "Flag_VM_DRS_Config");
```
Comment (m-brophy) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java]: Has something changed here? It looks to me like 5 rules fire but the assertion of 2 is successful. I don't understand.

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java [hunk 3]
```diff
@@ -374,6 +374,93 @@ public void test_CpuAffinityNotNull()
...
+        vmWorkloadInventoryModel.setHasEncryptedDisk(false);
+        vmWorkloadInventoryModel.setHasOpaqueNetwork(false);
+        facts.put("vmWorkloadInventoryModel", vmWorkloadInventoryModel);
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = new WorkloadInventoryReportModel();
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
+
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+        Assert.assertEquals(2, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener, "AgendaFocusForTest");
```
Comment (m-brophy) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java]: Again, the assertion of 2 rules firing shouldn't pass

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java [hunk 4]
```diff
@@ -374,6 +374,93 @@ public void test_CpuAffinityNotNull()
...
+        vmWorkloadInventoryModel.setHasEncryptedDisk(null);
+        vmWorkloadInventoryModel.setHasOpaqueNetwork(null);
+        facts.put("vmWorkloadInventoryModel", vmWorkloadInventoryModel);
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = new WorkloadInventoryReportModel();
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
 
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+        Assert.assertEquals(2, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener, "AgendaFocusForTest");
```
Comment (m-brophy) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java]: Assertion of 2 rules firing should not pass

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java (hunk 1 above)
Comment (jonathanvila) [reply src/main/java/org/jboss/xavier/analytics/pojo/output/workload/inventory/WorkloadInventoryReportModel.java]: :+1:

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java (hunk 2 above)
Comment (jonathanvila) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/FlagsTest.java]: You are totally right !!!! And the trick is the missing @Test annotation in the last 3 methods.
Good catch :) Thank you

**Review by m-brophy — APPROVED** [review]

**Pull request merged**